presto-pinot-toolkit/src/test/java/com/facebook/presto/pinot/query/TestPinotQueryGeneratorSql.java

```diff
@@ -158,6 +160,36 @@ public void testAggregationWithOrderByPushDownInTopN()
                 ImmutableMap.of());
     }
 
+    @Test(expectedExceptions = NoSuchElementException.class)
+    public void testLimitOrderByPushdown()
+    {
+        PlanBuilder planBuilder = createPlanBuilder(defaultSessionHolder);
+        TableScanNode tableScanNode = tableScan(planBuilder, pinotTable, city, fare);
+        AggregationNode agg = planBuilder.aggregation(
+                aggBuilder -> aggBuilder
+                        .source(tableScanNode)
+                        .singleGroupingSet(variable("city"))
+                        .addAggregation(planBuilder.variable("sum_fare"), getRowExpression("sum(fare)", defaultSessionHolder)));
+        int limitToPushdownOrderByForBrokerQueries = 5000;
+        pinotConfig.setLimitToPushdownOrderByForBrokerQueries(limitToPushdownOrderByForBrokerQueries);
+        TopNNode topN = new TopNNode(
+                planBuilder.getIdAllocator().getNextId(),
+                agg,
+                limitToPushdownOrderByForBrokerQueries,
+                new OrderingScheme(ImmutableList.of(new Ordering(variable("sum_fare"), SortOrder.ASC_NULLS_FIRST))),
+                TopNNode.Step.SINGLE);
+        testPinotQuery(
+                pinotConfig,
+                topN,
+                "SELECT city, sum(fare) FROM realtimeOnly GROUP BY city ORDER BY sum(fare) LIMIT " + limitToPushdownOrderByForBrokerQueries,
```

Comment:
Can you also test a non limited order by ? Which should perhaps be a SortNode ? Does that also have the same problem as a TopNNode ?

Reply:
Non limit orderby query will generate a SortNode, which is not pushdown-able, so it has no problem like this.

Reply:
Also there is a default limit applied to Pinot query/